kratos/utilities/variable_utils.h

```diff
@@ -183,16 +189,63 @@ class VariableUtils
      * @param SavedVariable: reference to the destination scalar variable
      * @param rNodes: reference to the objective node set
      */
-    void SaveScalarVar(const Variable< double >& OriginVariable,
-                       Variable< double >& SavedVariable,
-                       ModelPart::NodesContainerType& rNodes)
+    void SaveScalarVar(
+        const Variable< double >& OriginVariable,
+        Variable< double >& SavedVariable,
+        NodesContainerType& rNodes
+        )
     {
         KRATOS_TRY
         #pragma omp parallel for
-        for (int k = 0; k < static_cast<int> (rNodes.size()); k++)
+        for (int k = 0; k < static_cast<int> (rNodes.size()); ++k)
         {
-            ModelPart::NodesContainerType::iterator i = rNodes.begin() + k;
-            i->SetValue(SavedVariable,i->FastGetSolutionStepValue(OriginVariable));
+            NodesIteratorType it_node = rNodes.begin() + k;
+            it_node->SetValue(SavedVariable,it_node->FastGetSolutionStepValue(OriginVariable));
+        }
+        KRATOS_CATCH("")
+    }
+    
+    /**
+     * Takes the value of a non-historical vector variable and sets it in other non-historical variable
+     * @param OriginVariable: reference to the origin vector variable
+     * @param SavedVariable: reference to the destination vector variable
+     * @param rNodes: reference to the objective node set
+     */
+    void SaveVectorNonHistVar(
```

Comment:
I'd have that the variable is nodal in the method name (something like SaveVectorNonHistNodevar). However, this naming convention matches the non historical variable method above. Maybe we can discuss about renaming the existent ones.